src/server/api/routers/user.ts

```diff
@@ -46,4 +46,65 @@ export const userRouter = createTRPCRouter({
         },
       });
     }),
+
+  updateProfile: protectedProcedure
+    .input(
+      z
+        .object({
+          name: z.string().min(1),
+          emailAddress: z.string().email(),
+          dateOfBirth: z.string().min(1),
+          biography: z.string().min(1),
+          selectedUniversity: z.string().min(1),
+          topIndustries: z.array(z.string()).min(1),
+        })
+        .partial(),
+    )
+    .mutation(async ({ input, ctx }) => {
+      //Check if the user exits before trying to modify their information
+      const profileExits = await db.user.findUnique({
```

Comment:
profileExists :D

Reply:
We can do this check here, but
1. The ctx is populated directly from the database, so the user should never not exist.
2. If we don't check it here, the try catch you have around the the update statement will throw an error if there is no user.

So you can safely get rid of this check